src/openforms/prefill/base.py

```diff
@@ -44,11 +48,28 @@ def get_co_sign_values(
         The return value is a dict keyed by field name as specified in
         ``self.co_sign_fields``.
 
-        :param identfier: the unique co-signer identifier used to look up the details
+        :param identifier: the unique co-signer identifier used to look up the details
           in the pre-fill backend.
         :return: a key-value dictionary, where the key is the requested attribute and
           the value is the prefill value to use for that attribute.
         """
         raise NotImplementedError(
             "You must implement the 'get_co_sign_values' method."
         )  # pragma: nocover
+
+    def get_identifier_value(
+        self, submission: Submission, identifier_role: str
+    ) -> str | None:
+        """
+        Given a submission and the role of the identifier, return the value of the identifier.
+
+        The role of the identifier has to do with whether it is the 'main' identifier or an identifier
+        of someone logging in on behalf of someone/something else.
+
+        :param submission: an active :class:`Submission` instance
+        :param identifier_role: A string with one of the choices in :class:`IdentifierRoles`
+        :return: The value for the identifier
+        """
+        raise NotImplementedError(
```

Comment:
can we not provide a default implementation that has the old behaviour? This seems like a bit of an unnecessary breaking change.

Reply:
Do you mean that to change the base implementation with:
```python
        if not submission.is_authenticated:
            return

        if (
            identifier_role == IdentifierRoles.main
            and submission.auth_info.attribute == self.requires_auth
        ):
            return submission.auth_info.value
```
? 
(This method did not exist before the change, we would get the identifier inside `get_prefill_values`)

Reply:
I think that would work.

It's still highly dependent on the authentication plugin that's used though, but let's see if we can make things more agnostic or more formal in the next iteration(s) 🤔